clusterloader2/cmd/clusterloader.go

```diff
@@ -307,20 +306,55 @@ func main() {
 
 	testReporter := test.CreateSimpleReporter(path.Join(clusterLoaderConfig.ReportDir, "junit.xml"), "ClusterLoaderV2")
 	testReporter.BeginTestSuite()
+
+	var contexts []test.Context
+	var testConfigs []*api.Config
```

Comment:
It looks like there is a 1-1 relationship between a `testContext` and its `testConfig`. What's more, a `testConfig` can be created from a `testContext`. Given that, I'm not sure whether it makes sense to keep these two objects separately. We could either introduce a `ConfigWithContext` object or just add `testConfig` to the `Context`.  Thoughts?

Reply:
Adding `testConfig` to the `Context` makes more sense to me given that we have already moved `TestScenarios` in there. Or perhaps even added to TestScenario?

Reply:
Yeah, so let's do that (move testConfig to the context). Other than that, this PR looks great!